testing/buildFuzz.sh

```diff
@@ -0,0 +1,251 @@
+#	CXXFLAGS="$CFLAGS"
+#	LIB_FUZZING_ENGINE=-fsanitize=fuzzer
+#	
+#	# Run this script to build the fuzzers.
+#	./buildFuzz.sh
+#	
+#	# Run a fuzzer called someFuzzer.
+#	cd "$OUT"
+#	./someFuzzer optionalCorpusDir
+
+# Set default IFS, just for ensuring sanity.
+IFS=' 	
+'
+
+distributiveConcat() {
+	distributiveConcat_left="$1"
+	shift
+	for distributiveConcat_right in "$@"; do
+		printf '%s%s ' "$distributiveConcat_left" "$distributiveConcat_right"
+	done
+}
+
+mkdirCD() {
+	printf '%s\n' "$1"
+	test -d "$1" || mkdir "$1"
+	cd "$1"
+}
+
+# Arguments for $CC passed after $CFLAGS. The "-f" arguments are for
+# enabling nonstandard features that plan9port C code uses. Ideally
+# building with INSTALL would also use those flags. Even more ideally
+# those flag arguments would not be needed.
+cflags="-I $PLAN9/include -D PLAN9PORT -fsigned-char -fno-strict-aliasing -fno-strict-overflow -Wno-parentheses -Wno-comment"
+
+# Environment variables sometimes used for choosing translation units.
+. "$PLAN9/buildEnvironmentVariables.sh"
+
+test -e "$WORK" || mkdir "$WORK"
+test -e "$OUT" || mkdir "$OUT"
+
+cd "$WORK"
+
+target=lib9
+mkdirCD "$target"
+"$CC" $CFLAGS $cflags -c -I fmt "$PLAN9/src/$target/getcallerpc-$OBJTYPE."* `distributiveConcat "$PLAN9/src/$target/" _exits.c _p9dialparse.c _p9dir.c announce.c argv0.c atexit.c atoi.c atol.c atoll.c atnotify.c await.c cistrcmp.c cistrncmp.c cistrstr.c cleanname.c convD2M.c convM2D.c convM2S.c convS2M.c create.c crypt.c ctime.c dial.c dirfstat.c dirfwstat.c dirmodefmt.c dirread.c dirstat.c dirwstat.c dup.c encodefmt.c errstr.c exec.c execl.c exitcode.c fcallfmt.c get9root.c getenv.c getfields.c getnetconn.c getns.c getuser.c getwd.c jmp.c lrand.c lnrand.c main.c malloc.c malloctag.c mallocz.c nan.c needsrcquote.c needstack.c netcrypt.c netmkaddr.c notify.c nrand.c nulldir.c open.c opentemp.c pin.c pipe.c post9p.c postnote.c qlock.c quote.c rand.c read9pmsg.c readcons.c readn.c rfork.c searchpath.c seek.c sendfd.c sleep.c strdup.c strecpy.c sysfatal.c syslog.c sysname.c time.c tm2sec.c tokenize.c truerand.c u16.c u32.c u64.c unsharp.c wait.c waitpid.c write.c zoneinfo.c fmt/dofmt.c fmt/fltfmt.c fmt/fmt.c fmt/fmtfd.c fmt/fmtfdflush.c fmt/fmtlocale.c fmtlock2.c fmt/fmtnull.c fmt/fmtprint.c fmt/fmtquote.c fmt/fmtrune.c fmt/fmtstr.c fmt/fmtvprint.c fmt/fprint.c fmt/nan64.c fmt/print.c fmt/runefmtstr.c fmt/runeseprint.c fmt/runesmprint.c fmt/runesnprint.c fmt/runesprint.c fmt/runevseprint.c fmt/runevsmprint.c fmt/runevsnprint.c fmt/seprint.c fmt/smprint.c fmt/snprint.c fmt/sprint.c fmt/strtod.c fmt/vfprint.c fmt/vseprint.c fmt/vsmprint.c fmt/vsnprint.c fmt/charstod.c fmt/pow10.c utf/rune.c utf/runestrcat.c utf/runestrchr.c utf/runestrcmp.c utf/runestrcpy.c utf/runestrdup.c utf/runestrlen.c utf/runestrecpy.c utf/runestrncat.c utf/runestrncmp.c utf/runestrncpy.c utf/runestrrchr.c utf/runestrstr.c utf/runetype.c utf/utfecpy.c utf/utflen.c utf/utfnlen.c utf/utfrrune.c utf/utfrune.c utf/utfutf.c`
```

Comment:
Is there some way you can use `mk(1)` and the existing mkfiles instead of duplicating the filenames here?

Reply:
If you think of a way to do that, tell. The problem is that the existing build system is bent on ignoring user set CFLAGS and other environment variables.

I agree the current way is hackish, so my inclination is to auto-generate those parts of the build script using `mk -a -n all`.

Reply:
BTW, I am not sure if this is clear enough already: the current design of buildFuzz.sh (which environment variables it depends on) is guided by OSS-Fuzz.

See the oss-fuzz directory and the linked URIs if interested.

Reply:
@fhs Does the latest commit "￼testing: build libraries by hijacking mk's executables" resolve this issue? The C file filenames for plan9port libraries are no longer in buildFuzz.sh, instead I hijack 9c and 9ar from under `mk -a` by prepending to the PATH environment variable.

Reply:
Yes, looks good. Thanks.